integration-cli/docker_api_swarm_test.go

```diff
@@ -774,3 +777,108 @@ func setGlobalMode(s *swarm.Service) {
 		Global: &swarm.GlobalService{},
 	}
 }
+
+func checkClusterHealth(c *check.C, cl []*SwarmDaemon, managerCount, workerCount int) {
+	var totalMCount, totalWCount int
+	for _, d := range cl {
+		info, err := d.info()
+		c.Assert(err, check.IsNil)
+		if !info.ControlAvailable {
+			totalWCount++
+			continue
+		}
+		var leaderFound bool
+		totalMCount++
+		var mCount, wCount int
+		for _, n := range d.listNodes(c) {
```

Comment:
This API call gets proxied to the leader, so what we really see is the leader's view of the cluster.

I still think this loop is worth doing, because it confirms that API calls can be run against each of the managers. But it's important to understand that the information returned is really coming from the leader.